src/Map/src/MapOptionsNormalizer.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map;
+
+use Symfony\UX\Map\Bridge as MapBridge;
+use Symfony\UX\Map\Exception\UnableToDenormalizeOptionsException;
+use Symfony\UX\Map\Exception\UnableToNormalizeOptionsException;
+
+/**
+ * Normalizes and denormalizes map options.
+ *
+ * @internal
+ *
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+final class MapOptionsNormalizer
+{
+    private const string KEY_PROVIDER = '@provider';
+
+    /**
+     * @var array<string, class-string<MapOptionsInterface>>
+     */
+    public static array $providers = [
+        'google' => MapBridge\Google\GoogleOptions::class,
+        'leaflet' => MapBridge\Leaflet\LeafletOptions::class,
+    ];
+
+    public static function denormalize(array $array): MapOptionsInterface
+    {
+        if (null === ($provider = $array[self::KEY_PROVIDER] ?? null)) {
+            throw UnableToDenormalizeOptionsException::missingProviderKey(self::KEY_PROVIDER);
+        }
+
+        if (!isset(self::$providers[$provider])) {
+            throw UnableToDenormalizeOptionsException::unsupportedProvider($provider, array_keys(self::$providers));
+        }
+
+        unset($array[self::KEY_PROVIDER]);
+
+        $class = self::$providers[$provider];
+
+        return $class::fromArray($array);
+    }
+
+    public static function normalize(MapOptionsInterface $options): array
+    {
+        $provider = array_search($options::class, self::$providers, true);
+        if (!\is_string($provider)) {
+            throw UnableToNormalizeOptionsException::unsupportedProviderClass($options::class);
+        }
+
+        $array = $options->toArray();
+        $array[self::KEY_PROVIDER] = $provider;
+
+        return $array;
```

Comment:
```suggestion
        return [
            ...$options->toArray(), 
            self::KEY_PROVIDER => $provider,
        ];
```

Reply:
No need to create a new array again IMO

Reply:
It does not really ... 

I think anywhere you use merge or + or send the array back, spread is quicker. will check again :)

Reply:
You make me curious:
- if I read this code, we are creating a new array from a spreaded array and a new entry
- if I read my previous code, we re-use the array previously created and add a new entry to it

But maybe I'm wrong and biased due to how this feature works in the JavaScript world (OFC depending of the runtime etc, but I saw many projects improve performances by not using object spreading!).

Do you think an optimization is done by PHP?